.gitattributes

```diff
@@ -0,0 +1,132 @@
+# | `rust`     | Rust language.                                         |
+# | `tex`      | LaTeX documents.                                       |
+
+######################################################################
+
+# ====================================================================
+# MACROS #
+# ====================================================================
+# All macros assume the default rule is: `text eol=lf` and sets/removes attributes accordingly
+# If the default rule changes, these macros must be updated as well.
+
+# OVERRIDES
+[attr]binary        -text -diff -merge !eol
+
+# CUSTOM
+
+### Prevent end-of-line conversion
+[attr]input         -text -eol
+
+## Diff Macros
+### Useful for declaring which diff pattern to use for files that may
+### not be captured correctly by Git's heuristics.
+[attr]bash          diff=bash
+[attr]css           diff=css
+[attr]html          diff=html
+[attr]xml           diff=html
+[attr]markdown      diff=markdown
+[attr]python        diff=python
+[attr]ruby          diff=ruby
+
+### Do not generate a diff, instead print a "Binary files differ" message
+### Useful for autogenerated text files that may produce large, noisy diffs
+[attr]autogen       -diff
+[attr]lockfile      -diff
+
+### Fix syntax highlighting on GitHub to allow comments
+[attr]jsonc         linguist-language=JSON-with-Comments
+
+######################################################################
+
+# ====================================================================
+# RULES
+# ====================================================================
+
+# --------------------------------------------------------------------
+# DEFAULT RULE
+# --------------------------------------------------------------------
+# Auto detect text files and perform LF normalization
+*                       text eol=lf
+
+# --------------------------------------------------------------------
+# PATTERN RULES
+# --------------------------------------------------------------------
+# Common settings that generally should always be used with your language specific settings
+
+# Text files where line endings should be preserved
+*.patch    -text
+
+# Assets
+## Graphics
+*.gif           binary
+*.ico           binary
+*.jpeg          binary
+*.jpg           binary
+*.png           binary
+*.tif           binary
+*.tiff          binary
+*.webp          binary
+*.svgz          binary
+
+## Fonts
+*.ttf             binary
+*.eot             binary
+*.otf             binary
+*.woff            binary
+*.woff2           binary
+
+
+# These are explicitly windows files and should use crlf
+*.bat      eol=crlf
+*.cmd      text eol=crlf
+*.ps1      text eol=crlf
+
+
+# Uncommon (to Git) sourcecode files
+*.scss            text css
+
+
+# Fix syntax highlighting on GitHub to allow comments
+tsconfig.json     jsonc
+
+# Generated and/or minified files
+*.map             autogen
+*.lock            lockfile
```

Comment:
I feel like `Gemfile.lock` changes are small enough to understand and can be valuable, especially when debugging issues, WDYT about removing this so the diffs don't get collapsed always?

Reply:
I can definitely do that. These are just generic rules.

Reply:
Fixed! See the [diff](https://github.com/18F/identity-idp/pull/10518/commits/cc8f65b4d8ee6389ed89b1857d513bd0c47f7d5a#diff-618cd5b83d62060ba3d027e314a21ceaf75d36067ff820db126642944145393eR132-R134)